spec/models/df_json_wrapper_spec.rb

```diff
@@ -0,0 +1,47 @@
+require 'rails_helper'
+
+class DfJsonTest < DfJsonWrapper
+  def self.selectors = {
+    dict_one: { type: :boolean, key: "dictOfNumbers one" },
+    list_of_numbers: { type: :list, key: "listOfNumbers" }
+  }
+
+  define_json_readers
+end
+
+describe DfJsonWrapper do
+  let(:json) { JSON.parse(json_string) }
+  let(:instance) { DfJsonTest.new(json) }
+
+  describe "#define_json_readers" do
+    context "when the JSON has the correct structure" do
+      let(:json_string) {
+        <<~JSON_STRING
```

Comment:
NIT: In some editors, if you use the name of the format for the heredoc (eg, `JSON` here), it will engage the LSP and give goodies like syntax highlighting and even linting/completion

![image](https://github.com/user-attachments/assets/ebbcbd7b-fc67-4e62-8b7f-980a4a9c99d7)

Reply:
oooh, it works in rubymine too! thanks, that's awesome.